From f4ec208c07815bdc7c8e5bf4a724cea84ed2878a Mon Sep 17 00:00:00 2001
From: Yu Kuai <yukuai3@huawei.com>
Date: Wed, 12 Jul 2023 11:01:28 +0800
Subject: [PATCH 14/36] [Huawei] blk-io-hierarchy: disable hierarchy stats by
 default

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188826

--------------------------------

Commit b9585c8a738c ("[Huawei] blk-io-hierarchy: add Kconfig and
Makefile") enable hierarchy stats by default. In case some customer
don't want to use this, let's disable this by default.

Signed-off-by: Yu Kuai <yukuai3@huawei.com>
---
 arch/arm64/configs/openeuler_defconfig | 2 +-
 arch/sw_64/configs/openeuler_defconfig | 2 +-
 arch/x86/configs/openeuler_defconfig   | 2 +-
 block/Kconfig                          | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/configs/openeuler_defconfig b/arch/arm64/configs/openeuler_defconfig
index 8f76c3929942..9882b0df74f4 100644
--- a/arch/arm64/configs/openeuler_defconfig
+++ b/arch/arm64/configs/openeuler_defconfig
@@ -918,7 +918,7 @@ CONFIG_BLK_CGROUP_IOCOST=y
 CONFIG_BLK_CGROUP_LEGACY_IOCOST=y
 CONFIG_BLK_WBT_MQ=y
 CONFIG_BLK_DEBUG_FS=y
-CONFIG_BLK_IO_HIERARCHY_STATS=y
+# CONFIG_BLK_IO_HIERARCHY_STATS is not set
 CONFIG_BLK_DEBUG_FS_SWITCH=y
 CONFIG_BLK_DEBUG_FS_ZONED=y
 # CONFIG_BLK_SED_OPAL is not set
diff --git a/arch/sw_64/configs/openeuler_defconfig b/arch/sw_64/configs/openeuler_defconfig
index ae4346404583..0c44baad9264 100644
--- a/arch/sw_64/configs/openeuler_defconfig
+++ b/arch/sw_64/configs/openeuler_defconfig
@@ -532,7 +532,7 @@ CONFIG_BLK_DEV_INTEGRITY_T10=y
 # CONFIG_BLK_CGROUP_IOLATENCY is not set
 # CONFIG_BLK_CGROUP_IOCOST is not set
 CONFIG_BLK_DEBUG_FS=y
-CONFIG_BLK_IO_HIERARCHY_STATS=y
+# CONFIG_BLK_IO_HIERARCHY_STATS is not set
 CONFIG_BLK_DEBUG_FS_SWITCH=y
 # CONFIG_BLK_SED_OPAL is not set
 # CONFIG_BLK_INLINE_ENCRYPTION is not set
diff --git a/arch/x86/configs/openeuler_defconfig b/arch/x86/configs/openeuler_defconfig
index 09f2162a864e..56730bb5bb26 100644
--- a/arch/x86/configs/openeuler_defconfig
+++ b/arch/x86/configs/openeuler_defconfig
@@ -910,7 +910,7 @@ CONFIG_BLK_CGROUP_IOCOST=y
 CONFIG_BLK_CGROUP_LEGACY_IOCOST=y
 CONFIG_BLK_WBT_MQ=y
 CONFIG_BLK_DEBUG_FS=y
-CONFIG_BLK_IO_HIERARCHY_STATS=y
+# CONFIG_BLK_IO_HIERARCHY_STATS is not set
 CONFIG_BLK_DEBUG_FS_SWITCH=y
 CONFIG_BLK_DEBUG_FS_ZONED=y
 # CONFIG_BLK_SED_OPAL is not set
diff --git a/block/Kconfig b/block/Kconfig
index 12c9ef85a195..91bcc9b508b2 100644
--- a/block/Kconfig
+++ b/block/Kconfig
@@ -195,7 +195,7 @@ config BLK_DEBUG_FS
 
 config BLK_IO_HIERARCHY_STATS
 	bool "Enable hierarchy io stats"
-	default y
+	default n
 	depends on BLK_DEBUG_FS
 	help
 	Enabling this lets the block layer to record additional information
-- 
2.17.1

